### PR TITLE
feat: lock event types for org users 

### PR DESCRIPTION
**Pull request by sean-brydon**
fixes: CAL-3152 #13716 

Adds the ability for org customers to stop users creating events on their own `acme.cal.com/[user]` profile and only rely on managed event types

Loom: https://www.loom.com/share/9dd0f8704d7e42a0891f50deffb9bf17

### CODE REVIEW

File: apps/web/modules/event-types/views/event-types-listing-view.tsx
Comment (sean-brydon) [inline apps/web/modules/event-types/views/event-types-listing-view.tsx:899]: If locked and no managed event types - we dont need to show to show this "group"

File: apps/web/modules/event-types/views/event-types-listing-view.tsx
Comment (sean-brydon) [inline apps/web/modules/event-types/views/event-types-listing-view.tsx:928]: This last null fixes the bug where currently MEMBERS see the option to create events on teams that are empty even tho they dont have permission

File: packages/trpc/server/routers/viewer/eventTypes/create.handler.ts
Comment (CarinaWolli) [inline packages/trpc/server/routers/viewer/eventTypes/create.handler.ts:89]:
```suggestion
  if (ctx.user.organizationId && !ctx.user?.organization?.isOrgAdmin && !teamId) {
```

File: apps/web/modules/event-types/views/event-types-listing-view.tsx
Comment (CarinaWolli) [inline apps/web/modules/event-types/views/event-types-listing-view.tsx:925]: Do we not want to show the team event types anymore in read-only?

File: apps/web/modules/event-types/views/event-types-listing-view.tsx
Comment (CarinaWolli) [reply apps/web/modules/event-types/views/event-types-listing-view.tsx:899]: what about the hidden event types? shouldn't we still show them

Comment (CarinaWolli) [reply apps/web/modules/event-types/views/event-types-listing-view.tsx:899]: we should also lock the toggle of the hidden event types

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
Nice work 👏🏻 found some things that should be addressed before it can be approved 🙏

File: apps/web/modules/event-types/views/event-types-listing-view.tsx
Comment (sean-brydon) [reply apps/web/modules/event-types/views/event-types-listing-view.tsx:925]: This is only the empty list - it has a cta to create events (when it’s read only the user doesn’t have permission to do so) and the user can start the flow and hit an error 

so this is just hiding it from their view - the event list itself is still rendered

File: apps/web/modules/event-types/views/event-types-listing-view.tsx
Comment (sean-brydon) [reply apps/web/modules/event-types/views/event-types-listing-view.tsx:899]: Good spot! I always had a managed event in there when testing :P

File: apps/web/modules/event-types/views/event-types-listing-view.tsx
Comment (CarinaWolli) [reply apps/web/modules/event-types/views/event-types-listing-view.tsx:925]: Oh yeah sorry, somehow it also didn't show the read-only event types for me. But works now 🙏

**Comment by sean-brydon** [comment]
@CarinaWolli feedback has been addressed please re-review <3

File: apps/web/modules/event-types/views/event-types-listing-view.tsx
Comment (CarinaWolli) [inline apps/web/modules/event-types/views/event-types-listing-view.tsx:899]: > - what about the hidden event types? shouldn't we still show them
> - we should also lock the toggle of the hidden event types

I don't see any changes here, these two things are still an issue for me. 

Example: Admin locks event types for members. Member has one personal event type (no managed event type). Then we want to mark the event type as hidden and it should still show in the event type list. Right now the hidden event types are not shown if the user doesn't have a managed event type. Also the toggle to make the event type public again should be disabled for the member

Comment (sean-brydon) [reply apps/web/modules/event-types/views/event-types-listing-view.tsx:899]: I think I forgot to push some code 😅

Comment (sean-brydon) [reply apps/web/modules/event-types/views/event-types-listing-view.tsx]: Commit history looks weird as the feedback was timestamped before this message :P 

Latest changes should be there

File: apps/web/modules/event-types/views/event-types-listing-view.tsx
Comment (CarinaWolli) [inline apps/web/modules/event-types/views/event-types-listing-view.tsx:814]: We need to disable the `CreateButton` if we don't have any profileOptions. Now I am getting an error when clicking on the 'New' button `Uncaught TypeError: Cannot read properties of undefined (reading 'slug')`

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
Found two more small things that are blocking 🙏

We need to disable the toggle inside event type settings too
 Screenshot 2024-03-11 at 11 22 51 AM

File: apps/web/modules/event-types/views/event-types-listing-view.tsx
Comment (sean-brydon) [reply apps/web/modules/event-types/views/event-types-listing-view.tsx:814]: Never even thought of this - nice spot

**Review by CarinaWolli — DISMISSED** [review]
Looking good now, nice work 👏🏻

**Comment by CarinaWolli** [comment]
@sean-brydon can you fix the merge conflicts?

**Review by CarinaWolli — APPROVED** [review]
re-approve after fixing merge conflicts

**Pull request merged**
